### PR TITLE
CMPT-33308 more examples

### PR DESCRIPTION
**Pull request by volosatik**
CMPT-33557 floatingip

### CODE REVIEW

File: examples/networking/floatingip/main-with-port.tf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+resource "vkcs_networking_floatingip" "fip-with-port" {
```
Comment (ftersin) [inline examples/networking/floatingip/main-with-port.tf:1]: why do you need all this files above?

File: templates/networking/resources/vkcs_networking_floatingip.md.tmpl [hunk 2]
```diff
@@ -18,7 +18,11 @@ description: |-
 If one of the subnets in a list has an exhausted pool, terraform will try the
 next subnet ID from the list.
 
-{{tffile "templates/networking/resources/vkcs_networking_floatingip/subnet_list/main.tf"}}
+{{tffile "examples/networking/floatingip/main-subnets.tf"}}
```
Comment (ftersin) [inline templates/networking/resources/vkcs_networking_floatingip.md.tmpl:21]: i am not sure we want this case

File: examples/networking/floatingip/base-port.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+resource "vkcs_networking_port" "port-for-fip" {
```
Comment (ftersin) [inline examples/networking/floatingip/base-port.tf:1]: why do not use port example?

File: examples/networking/floatingip_associate/main.tf [hunk 4]
```diff
@@ -0,0 +1,7 @@
+resource "vkcs_networking_floatingip_associate" "floatingip-associate" {
+  floating_ip = vkcs_networking_floatingip.floatingip.address
+  port_id     = vkcs_networking_port.port.id
+  # ensure the router interface is up
+  # ensure that router is connected to external network
```
Comment (ftersin) [inline examples/networking/floatingip_associate/main.tf:5]: We do not ensure this here. If you want to mention this - change the resource description

File: examples/networking/floatingip_associate/main.tf [hunk 5]
```diff
@@ -0,0 +1,7 @@
+resource "vkcs_networking_floatingip_associate" "floatingip-associate" {
+  floating_ip = vkcs_networking_floatingip.floatingip.address
+  port_id     = vkcs_networking_port.port.id
+  # ensure the router interface is up
```
Comment (ftersin) [inline examples/networking/floatingip_associate/main.tf:4]: Start a sentence with a capital letter

File: examples/networking/floatingip_associate/base-port.tf [hunk 6]
```diff
@@ -0,0 +1,7 @@
+resource "vkcs_networking_port" "port" {
```
Comment (ftersin) [inline examples/networking/floatingip_associate/base-port.tf:1]: do not we want to use port example?

File: examples/networking/port/main-datasource.tf [hunk 7]
```diff
@@ -0,0 +1,7 @@
+data "vkcs_networking_port" "persistent_etcd" {
+  tags       = ["tf-example"]
```
Comment (ftersin) [inline examples/networking/port/main-datasource.tf:2]: lets add another one tag - etcd

File: templates/networking/resources/vkcs_networking_floatingip.md.tmpl (hunk 2 above)
Comment (volosatik) [reply templates/networking/resources/vkcs_networking_floatingip.md.tmpl:21]: im pretty sure that its a good example but may be with not pre-created pool and network. the user should be able to specify in which range he wants a floatingip so specifying subnet or list of subnets can help him achieve this. 
but then i will nee to create a pool here with several subnets

File: examples/networking/floatingip/main.tf [hunk 8]
```diff
@@ -0,0 +1,4 @@
+resource "vkcs_networking_floatingip" "floatingip" {
```
Comment (ftersin) [inline examples/networking/floatingip/main.tf:1]: base_fip

File: examples/networking/subnet/main-datasource.tf [hunk 9]
```diff
@@ -0,0 +1,4 @@
+data "vkcs_networking_subnet" "subnet_one_of_external" {
+  cidr       = "85.192.32.0/22"
+  network_id = data.vkcs_networking_network.extnet.id
```
Comment (ftersin) [inline examples/networking/subnet/main-datasource.tf:3]: use one of our subnets (app / db)

File: examples/networking/port_secgroup_associate/main.tf [hunk 10]
```diff
@@ -0,0 +1,6 @@
+resource "vkcs_networking_port_secgroup_associate" "port_secgroup_associate" {
```
Comment (ftersin) [inline examples/networking/port_secgroup_associate/main.tf:1]: append_secgroups

File: examples/networking/port_secgroup_associate/main-replace.tf [hunk 11]
```diff
@@ -0,0 +1,7 @@
+resource "vkcs_networking_port_secgroup_associate" "port_secgroup_associate_replace" {
```
Comment (ftersin) [inline examples/networking/port_secgroup_associate/main-replace.tf:1]: replace_secgroups

File: examples/networking/port_secgroup_associate/main-remove.tf [hunk 12]
```diff
@@ -0,0 +1,5 @@
+resource "vkcs_networking_port_secgroup_associate" "port_secgroup_associate_remove" {
```
Comment (ftersin) [inline examples/networking/port_secgroup_associate/main-remove.tf:1]: remove_secgroups

File: examples/networking/floatingip/base-port.tf (hunk 3 above)
Comment (volosatik) [reply examples/networking/floatingip/base-port.tf]: used

File: examples/networking/floatingip_associate/main.tf (hunk 4 above)
Comment (volosatik) [reply examples/networking/floatingip_associate/main.tf]: changed

File: examples/networking/floatingip_associate/main.tf (hunk 5 above)
Comment (volosatik) [reply examples/networking/floatingip_associate/main.tf]: used

File: examples/networking/floatingip_associate/base-port.tf (hunk 6 above)
Comment (volosatik) [reply examples/networking/floatingip_associate/base-port.tf]: used

File: examples/networking/port/main-datasource.tf (hunk 7 above)
Comment (volosatik) [reply examples/networking/port/main-datasource.tf]: added

File: templates/networking/resources/vkcs_networking_floatingip.md.tmpl [hunk 13]
```diff
@@ -18,7 +18,11 @@ description: |-
...
+
+### Floating IP allocation with connection to port
```
[the remark this answers is not in the document]
Comment (volosatik) [reply templates/networking/resources/vkcs_networking_floatingip.md.tmpl]: renamed

File: examples/networking/floatingip/main.tf (hunk 8 above)
Comment (volosatik) [reply examples/networking/floatingip/main.tf]: renamed

File: examples/networking/floatingip/main-datasource.tf [hunk 14]
```diff
@@ -0,0 +1,6 @@
+data "vkcs_networking_floatingip" "fip-with-port" {
```
[the remark this answers is not in the document]
Comment (volosatik) [reply examples/networking/floatingip/main-datasource.tf]: renamed

File: examples/networking/subnet/main-datasource.tf (hunk 9 above)
Comment (volosatik) [reply examples/networking/subnet/main-datasource.tf]: used

File: examples/networking/port_secgroup_associate/main.tf (hunk 10 above)
Comment (volosatik) [reply examples/networking/port_secgroup_associate/main.tf]: renamed

File: examples/networking/port_secgroup_associate/main-replace.tf (hunk 11 above)
Comment (volosatik) [reply examples/networking/port_secgroup_associate/main-replace.tf]: renamed

File: examples/networking/port_secgroup_associate/main-remove.tf (hunk 12 above)
Comment (volosatik) [reply examples/networking/port_secgroup_associate/main-remove.tf]: renamed

File: examples/networking/floatingip/main-with-port.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (volosatik) [reply examples/networking/floatingip/main-with-port.tf]: renamed

File: templates/networking/resources/vkcs_networking_floatingip_associate.md.tmpl [hunk 15]
```diff
@@ -11,6 +11,7 @@ description: |-
 {{ .Description }}
 
 ## Example Usage
+## Make sure that router is connected to external network
```
Comment (ftersin) [inline templates/networking/resources/vkcs_networking_floatingip_associate.md.tmpl:14]: this must be in another place

File: examples/networking/port/main-datasource.tf [hunk 16]
```diff
@@ -0,0 +1,7 @@
+data "vkcs_networking_port" "persistent_etcd" {
+  tags       = ["tf-example","etcd"]
```
Comment (ftersin) [inline examples/networking/port/main-datasource.tf:2]: should not be this formatted to add space after comma?

File: templates/networking/resources/vkcs_networking_port_secgroup_associate.md.tmpl [hunk 17]
```diff
@@ -12,11 +12,11 @@ description: |-
 
 ## Example Usage
 ### Append a security group to an existing port
-{{tffile "templates/networking/resources/vkcs_networking_port_secgroup_associate/append/main.tf"}}
+{{tffile "examples/networking/port_secgroup_associate/main.tf"}}
```
Comment (ftersin) [inline templates/networking/resources/vkcs_networking_port_secgroup_associate.md.tmpl:15]: could we use .ExampleFile here?

**Review by ftersin — APPROVED** [review]

**Pull request merged**